conbench/db.py

```diff
@@ -38,7 +38,11 @@ def configure_engine(url):
         # tickets and discussions:
         # https://github.com/conbench/conbench/issues/599
         # https://github.com/conbench/conbench/pull/690
-        connect_args={"options": "-c timezone=utc -c statement_timeout=60s"},
+        # https://docs.sqlalchemy.org/en/20/core/engines.html#use-the-connect-args-dictionary-parameter
+        connect_args={
+            "options": "-c timezone=utc -c statement_timeout=60s",
+            "connect_timeout": 3,
```

Comment:
Is this 3 seconds? Or some other unit? Not a big deal, just curious (I did look in the URL up above, but couldn't find what I was looking for either with a quick scan)

Reply:
Yes, unit is seconds!

For the record, the `connect_timeout` arg is documented here: https://www.postgresql.org/docs/12/libpq-connect.html

> Maximum wait for connection, in seconds (write as a decimal integer, e.g., 10). Zero, negative, or not specified means wait indefinitely. The minimum allowed timeout is 2 seconds, therefore a value of 1 is interpreted as 2. This timeout applies separately to each host name or IP address. For example, if you specify two hosts and connect_timeout is 5, each host will time out if no connection is made within 5 seconds, so the total time spent waiting for a connection might be up to 10 seconds.


What is maybe not obvious that the Python package psycopg2 is a wrapper around libpq, and libpq can be in client or server mode.

Reply:
Would you mind adding a small comment? Agreed with Jon that it's hard to find.

```suggestion
            "connect_timeout": 3,  # seconds
```

Reply:
Yes! Adding a commit.